### ISSUE TITLE
Token not found it the cache

### ISSUE THREAD

**Issue by macfisher** [Azure-Samples/ms-identity-java-webapi#5]
Having a number of issues with these code samples, but I've finally gotten the web client service in state that user can login via AD and successfully make a request to the OBO server with a valid JWT token.

The OBO service handles the request through MsalAuthHelper up until it gets into `AcquireTokenSilentSupplier.execute`. Code execution hits the following logic:
`if(silentRequest.parameters().account() == null){
            res = clientApplication.tokenCache.getCachedAuthenticationResult(
                    requestAuthority,
                    silentRequest.parameters().scopes(),
                    clientApplication.clientId());
        }`

At no point in execution do we have accessTokens stored within any reference to cache, and the following exception eventually gets thrown:

`com.microsoft.aad.msal4j.MsalClientException: Token not found it the cache
	at com.microsoft.aad.msal4j.AcquireTokenSilentSupplier.execute(AcquireTokenSilentSupplier.java:58) ~[msal4j-1.3.0.jar:1.3.0]
	at com.microsoft.aad.msal4j.AuthenticationResultSupplier.get(AuthenticationResultSupplier.java:59) ~[msal4j-1.3.0.jar:1.3.0]
	at com.microsoft.aad.msal4j.AuthenticationResultSupplier.get(AuthenticationResultSupplier.java:17) ~[msal4j-1.3.0.jar:1.3.0]
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run$$$capture(CompletableFuture.java:1700) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177) ~[na:na]`

I imagine this has to do with a misconfiguration, but there is very little to go off of here. What would cause the Authentication result to not store a cached accessToken? This also may be masking a bigger problem, like not having an accessToken? I've googled a lot about this specific issue, but all that I've read are the various tutorials and API docs, which I've read them all and cannot seem to figure out what is misconfigured or what to look at beyond the `scope` property that is ultimately passed in.

Also note that the code sample hard codes the following line in the `/graphMeApi` endpoint:
`String oboAccessToken = msalAuthHelper.getOboToken("https://graph.microsoft.com/.default");`

I've been using that, but also have switched it to use my API's specific `access_as_user` etc., but nothing I do to change that input makes any difference.

Any help you suggestions would be appreciated. I'd be happy to also submit a PR once I get this working because I've had to make changes here and there, and also the docs are lacking in regards to setting up each of the AD app Manifest files. Would love to contribute! Please help!

**Comment by sangonzal** [comment]
@macfisher You are right, the sample is broken in it's current state. 

Pull request #4 bumped up MSAL version to v1.3, but did not update the sample's code. If you change MSAL version to 0.0.5-preview, the sample should work. 

We'd happily take PR to update the sample to work with MSAL v1.3. Two changes that are needed are (might need more):
- Update `ClientCredentialFactory.from` to ClientCredentialFactory.fromSecret()
- Wrap the silent call where you are getting the exception in a try catch (instead of checking if IAuthenticationResult is null) If the exception is instance of MsalException, try with OBO. Otherwise, rethrow. Something along the lines of [this sample](https://github.com/AzureAD/microsoft-authentication-library-for-java/blob/dev/src/samples/confidential-client/ClientCredentialGrant.java)
 
FYI @SomkaPe @navyasric

**Comment by macfisher** [comment]
@sangonzal Thank you! I can confirm that the sample works now after reverting to `0.5.0-preview` version.

**Comment by kxying-kk** [comment]
@macfisher Hi. Mackenzie. I could not get "Call obo API" button work. I checked configuration in azure many times and did not find anything that i missed. 

Do you still remember any configurations missing from readme?

thanks
Kelly

**Comment by macfisher** [comment]
@kxying-kk What kind of errors are you seeing? Could you provide an error message or stack trace?

**Comment by kxying-kk** [comment]
@macfisher Here is the issue that i am facing. i was able to login after i changed http to https in redirect URL in azure to make the login work.. When I clicked “Call Obo Api” button, I got the following exception in web application(msal-web-sample).  The request was thrown on class AuthHelper.java method getAuthResultBySilentFlow. line: updatedResult = future.get(); obo application(msal-obo-sample) did not even get the request.

2020-03-22 15:17:05.967 DEBUG 14396 --- [nio-8080-exec-1] o.s.web.servlet.DispatcherServlet : GET "/obo_api", parameters={}
2020-03-22 15:17:05.968 DEBUG 14396 --- [nio-8080-exec-1] s.w.s.m.m.a.RequestMappingHandlerMapping : Mapped to public org.springframework.web.servlet.ModelAndView com.microsoft.azure.msalwebsample.AuthPageController.callOboApi(javax.servlet.http.HttpServletRequest) throws java.lang.Throwable
2020-03-22 15:17:06.294 INFO 14396 --- [onPool-worker-2] com.microsoft.aad.msal4j.HttpHelper : [Correlation ID: 166dac79-de19-46a2-8bef-a321a40fa338] Sent (166dac79-de19-46a2-8bef-a321a40fa338) Correlation Id is not same as received (null).
2020-03-22 15:17:06.295 ERROR 14396 --- [onPool-worker-2] c.m.a.m.ConfidentialClientApplication : [Correlation ID: 166dac79-de19-46a2-8bef-a321a40fa338] Execution of class com.microsoft.aad.msal4j.AcquireTokenSilentSupplier failed.

com.microsoft.aad.msal4j.MsalInteractionRequiredException: AADSTS70000: The request was denied because one or more scopes requested are unauthorized or expired. The user must first sign in and grant the client application access to the requested scope.
Trace ID: 490593c9-cb73-499c-b824-36ddc62a0200
Correlation ID: 166dac79-de19-46a2-8bef-a321a40fa338
Timestamp: 2020-03-22 19:17:07Z
at com.microsoft.aad.msal4j.MsalServiceExceptionFactory.fromHttpResponse(MsalServiceExceptionFactory.java:37) ~[msal4j-1.3.0.jar:1.3.0]
at com.microsoft.aad.msal4j.TokenRequestExecutor.createAuthenticationResultFromOauthHttpResponse(TokenRequestExecutor.java:81) ~[msal4j-1.3.0.jar:1.3.0]
at com.microsoft.aad.msal4j.TokenRequestExecutor.executeTokenRequest(TokenRequestExecutor.java:36) ~[msal4j-1.3.0.jar:1.3.0]
at com.microsoft.aad.msal4j.ClientApplicationBase.acquireTokenCommon(ClientApplicationBase.java:92) ~[msal4j-1.3.0.jar:1.3.0]
at com.microsoft.aad.msal4j.AcquireTokenByAuthorizationGrantSupplier.execute(AcquireTokenByAuthorizationGrantSupplier.java:52) ~[msal4j-1.3.0.jar:1.3.0]
at com.microsoft.aad.msal4j.AcquireTokenSilentSupplier.execute(AcquireTokenSilentSupplier.java:50) ~[msal4j-1.3.0.jar:1.3.0]
at com.microsoft.aad.msal4j.AuthenticationResultSupplier.get(AuthenticationResultSupplier.java:59) [msal4j-1.3.0.jar:1.3.0]
at com.microsoft.aad.msal4j.AuthenticationResultSupplier.get(AuthenticationResultSupplier.java:17) [msal4j-1.3.0.jar:1.3.0]
at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1604) [na:1.8.0_241]
at java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1596) [na:1.8.0_241]
at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) [na:1.8.0_241]
at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056) [na:1.8.0_241]
at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692) [na:1.8.0_241]
at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157) [na:1.8.0_241]

thanks
Kelly
